### PR TITLE
Support to grpc mocks

### PR DESCRIPTION
**Pull request by Vorobeyko**
This Pull Request added support for grpc with  https://github.com/ozonru/mountebank-grpc.

### CODE REVIEW

**Comment by brunns** [comment]
Nice. It does feel like these new attributes could be in subclasses, though - and maybe even a separate package eventually.

**Comment by Vorobeyko** [comment]
> Nice. It does feel like these new attributes could be in subclasses, though - and maybe even a separate package eventually.

If create subclasses for attributes, then it should look like this?:
```
class GrpcImposter(Imposter):
   pass
```

**Comment by brunns** [comment]
> > Nice. It does feel like these new attributes could be in subclasses, though - and maybe even a separate package eventually.
> 
> If create subclasses for attributes, then it should look like this?:
> 
> ```
> class GrpcImposter(Imposter):
>    pass
> ```

Something like this, I think:

```pythoin
class GrpcImpostor(Imposter):
    class Protocol(Enum):
        """Imposter `Protocol <http://www.mbtest.org/docs/protocols/http>`_."""
        GRPC = "grpc"

    def __init__(
            self,
            stubs: Union[Stub, Iterable[Stub]],
            port: Optional[int] = None,
            name: Optional[str] = None,
            record_requests: bool = True,
            options: "Options" = None,
            services: "Services" = None
    ) -> None:
        super(GrpcImpostor, self).__init__(stubs=stubs, port=port, protocol=GrpcImpostor.Protocol.GRPC, name=name, record_requests=record_requests)
        self.options = options
        self.services = services

    def as_structure(self) -> JsonStructure:
        structure = super().as_structure()
        if self.services:
            structure["services"] = self.services.as_structure()
        if self.options:
            structure["options"] = self.options.as_structure()
        return structure

    @staticmethod
    def from_structure(structure: JsonStructure) -> "Imposter":
        imposter = super().from_structure(structure)
        if "services" in structure:
            imposter.services = Services.from_structure(structure["services"])
        if "options" in structure:
            imposter.options = Options.from_structure(structure["options"])
        return imposter
```

Leaving the original `Impostor` class as it was. It would be nice to have some integration tests for it, so I can make suggestions which actually work. :-)

File: src/mbtest/imposters/imposters.py [hunk 1]
```diff
@@ -197,3 +206,51 @@ def smtp_imposter(name="smtp", record_requests=True) -> Imposter:
     return Imposter(
         [], 4525, protocol=Imposter.Protocol.SMTP, name=name, record_requests=record_requests
     )
+
+
+class ProtobufOptions:
```
Comment (brunns) [inline src/mbtest/imposters/imposters.py:211]: These classes should be subclasses of `JsonSerializable`, I think.

**Comment by brunns** [comment]
This looks really interesting, but I can't quite see what an integration test would look like for this. Have you got a simple example?

**Pull request closed without merging**